code/data.py

```diff
@@ -218,6 +218,35 @@ def load_significant_numbers():
             except ValueError:
                 sys.stderr.write("WARNING: Invalid number in 'numbers.dat' line: %d\n" % index)
 
+def load_internal_id():
+    internal_id_file = dirs.get_readable_file_in_dirs("internal_id.dat", "data")
+
+    with open(internal_id_file, 'r', encoding='utf-8') as file:
+        for index, line in enumerate(file):
+            line = line.strip()
+
+            if len(line) == 0 or line[0] == "#":
+                continue
+
+            try:
+                obj, internal_id   = line.split("=")
+                obj_type, obj_id  = obj.split("|")
+                
+                internal_id = internal_id.strip()
+                obj_type = obj_type.strip()
+                obj_id = obj_id.strip()
+
+                if not obj_type in g.internal_id_forward:
+                    g.internal_id_forward[obj_type] = {}
+                    g.internal_id_backward[obj_type] = {}
+
+                g.internal_id_forward[obj_type][obj_id] = internal_id
+                g.internal_id_backward[obj_type][internal_id] = obj_id
```

Comment:
Should assert that no key are reused? It would ensure that we catch bugs in the data file immediately rather than later.

Note that the tests cannot do that because the data structure would already be corrupted by the time the test is run (unless the tests have its own parser; which has its own issues).